internal/test/framework/windowsvm.go

```diff
@@ -181,53 +181,58 @@ func (w *windowsVM) RetrieveFiles(remoteDir, localDir string) error {
 	// Get the list of all files in the directory
 	remoteFiles, err := sftp.ReadDir(remoteDir)
 	if err != nil {
-		return fmt.Errorf("error opening remote file: %v", err)
+		return fmt.Errorf("error opening remote directory %s: %v", remoteDir, err)
 	}
 
 	for _, remoteFile := range remoteFiles {
-		// Assumption: We ignore the directories here the reason being RetrieveFiles should just retrieve files
-		// in a directory, if this is directory, we should have called RetrieveFiles on this directory
+		remotePath := filepath.Join(remoteDir, remoteFile.Name())
+		localPath := filepath.Join(localDir, remoteFile.Name())
+		// check if it is a directory, call itself again
 		if remoteFile.IsDir() {
-			continue
-		}
-		fileName := remoteFile.Name()
-		dstFile, err := os.Create(filepath.Join(localDir, fileName))
-		if err != nil {
-			log.Printf("error creating file locally: %v", err)
-			continue
-		}
-		// TODO: Check if there is some performance implication of multiple Open calls.
-		srcFile, err := sftp.Open(remoteDir + "\\" + fileName)
-
-		if err != nil {
-			log.Printf("error while opening remote directory on the Windows VM: %v", err)
-			continue
-		}
-		_, err = io.Copy(dstFile, srcFile)
-		if err != nil {
-			log.Printf("error retrieving file %v from Windows VM: %v", fileName, err)
-			continue
-		}
-		// flush memory
-		if err = dstFile.Sync(); err != nil {
-			log.Printf("error flusing memory: %v", err)
-			continue
-		}
-		if err := srcFile.Close(); err != nil {
-			log.Printf("error closing file on the remote host %s", fileName)
-			continue
-		}
-		if err := dstFile.Close(); err != nil {
-			log.Printf("error closing file %s locally", fileName)
-			continue
+			if err = w.RetrieveDirectories(remotePath, localPath); err != nil {
```

Comment:
You are passing `localPath` which in this case is a directory, so won't a new local directory get created? Or am I not following the code correctly?

Reply:
mkdirall makes a new local directory using parameter localDir which is the original argument, I am modifying and sending the localPath and not the localDir here.